src/subscription_manager/managercli.py

```diff
@@ -1166,6 +1168,7 @@ def _do_command(self):
                                  "by the server. Did not complete your request."))
             try:
                 attach.AttachService(self.cp).attach_auto(self.options.service_level)
+                set_intent_service_level(self.options.service_level)
```

Comment:
There should be only one line between try-except. I would change the code to something like this:

```python
try:
    attach.AttachService(self.cp).attach_auto(self.options.service_level)
except connection.RestlibException as re:
    print_error(re.msg)
except Exception:
    log.exception("Auto-attach failed")
    raise
else:
    # This block of code is executed, when no exception is raised
    set_intent_service_level(self.options.service_level)
    if self.options.service_level is not None:
        print(_("Service level set to: %s") % self.options.service_level)

```